pkg/provider/onepassword/onepassword.go

```diff
@@ -580,6 +587,56 @@ func (provider *ProviderOnePassword) getAllForVault(vaultID string, ref esv1beta
 	return nil
 }
 
+// waitForFunc will wait for OnePassword to _actually_ create/update the secret. OnePassword returns immediately after
+// the initial create/update which makes the next call for the same item create/update a new item with the same name. Hence, we'll
+// wait for the item to exist or be updated on OnePassword's side as well.
+// Ideally we could do bulk operations and handle data with one submit, but that would require re-writing the entire
+// push secret controller. For now, this is sufficient.
+func (provider *ProviderOnePassword) waitForFunc(ctx context.Context, fn func() error) error {
+	// check every .5 seconds
+	tick := time.NewTicker(500 * time.Millisecond)
+	defer tick.Stop()
+	done, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	for {
+		select {
+		case <-tick.C:
+			if err := fn(); err == nil {
+				return nil
+			}
+		case <-done.Done():
+			return fmt.Errorf("timeout to wait for function to run successful")
```

Comment:
nit, it could be informative if we return the later error we received

Reply:
+1

Reply:
ah you mean in the tick. sorry. :D

Reply:
Ah sorry, I meant "last" not "later"...